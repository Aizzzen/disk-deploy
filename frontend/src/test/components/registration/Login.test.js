import {fireEvent, render, screen} from "@testing-library/react";
import Login from "../../../components/registration/Login";
import {Provider} from "react-redux";
import {store} from "../../../store/store";

describe('Login component tests',  () => {
    test('Main tests', () => {
        render(
            <Provider store={store}>
                <Login/>
            </Provider>
        )
        const containerDiv = screen.getByTestId('login-page')
        const headerDiv = screen.getByText(/authorization/i)
        const emailInput = screen.getByPlaceholderText(/Enter your email/i)
        const passInput = screen.getByPlaceholderText(/Enter your password/i)
        const btn = screen.getByRole('button')

        expect(containerDiv).toBeInTheDocument()
        expect(headerDiv).toBeInTheDocument()
        expect(emailInput).toBeInTheDocument()
        expect(passInput).toBeInTheDocument()
        expect(btn).toBeInTheDocument()

        expect(btn).toMatchSnapshot()
        expect(headerDiv).toMatchSnapshot()
        expect(emailInput).toMatchSnapshot()
        expect(passInput).toMatchSnapshot()
        expect(btn).toMatchSnapshot()
    })

    test('Login events', () => {
        render(
            <Provider store={store}>
                <Login/>
            </Provider>
        )
        const input = screen.getByPlaceholderText(/Enter your email/i)
        expect(input).toContainHTML('')
        fireEvent.input(input, {
            target: {value: 'user@mail.ru'}
        })
        expect(input).toContainHTML('user@mail.ru')

        const passInput = screen.getByPlaceholderText(/Enter your password/i)
        expect(passInput).toContainHTML('')
        fireEvent.input(passInput, {
            target: {value: '123456'}
        })
        expect(passInput).toContainHTML('123456')
    })

});
