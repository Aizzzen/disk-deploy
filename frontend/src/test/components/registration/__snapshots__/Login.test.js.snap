// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests Main tests 1`] = `
<button
  class="authorization__btn"
>
  Login
</button>
`;

exports[`Login component tests Main tests 2`] = `
<div
  class="authorization__header"
>
  Authorization
</div>
`;

exports[`Login component tests Main tests 3`] = `
<input
  placeholder="Enter your email"
  type="text"
  value=""
/>
`;

exports[`Login component tests Main tests 4`] = `
<input
  placeholder="Enter your password"
  type="password"
  value=""
/>
`;

exports[`Login component tests Main tests 5`] = `
<button
  class="authorization__btn"
>
  Login
</button>
`;
